lib/improver/utilities/temporal.py

```diff
@@ -182,3 +190,179 @@ def forecast_period_coord(
            "coordinate were also not available for calculating "
            "the forecast_period.".format(cube))
     raise CoordinateNotFoundError(msg)
+
+
+def iris_time_to_datetime(time_coord):
+    """
+    Convert iris time to python datetime object. Working in UTC.
+
+    Args:
+        time_coord (iris.coord.Coord):
+            Iris time coordinate element(s).
+
+    Returns:
+        list of datetime.datetime objects
+            The time element(s) recast as a python datetime object.
+
+    """
+    time_coord.convert_units('seconds since 1970-01-01 00:00:00')
+    return [datetime.utcfromtimestamp(value) for value in time_coord.points]
+
+
+def dt_to_utc_hours(dt_in):
+    """
+    Convert python datetime.datetime into hours since 1970-01-01 00Z.
+
+    Args:
+        dt_in (datetime.datetime object):
+            Time to be converted.
+    Returns:
+        float:
+            hours since epoch
+
+    """
+    utc_seconds = mktime(dt_in.utctimetuple())
+    return utc_seconds/3600.
+
+
+def datetime_constraint(time_in, time_max=None):
+    """
+    Constructs an iris equivalence constraint from a python datetime object.
+
+    Args:
+        time_in (datetime.datetime object):
+            The time to be used to build an iris constraint.
+        time_max (datetime.datetime object):
+            Optional max time, which if provided leads to a range constraint
+            being returned up to < time_max.
+
+    Returns:
+        time_extract (iris.Constraint):
+            An iris constraint to be used in extracting data at the given time
+            from a cube.
+
+    """
+    time_start = PartialDateTime(
+        time_in.year, time_in.month, time_in.day, time_in.hour)
+
+    if time_max is None:
+        time_extract = Constraint(time=time_start)
+    else:
+        time_limit = PartialDateTime(
+            time_max.year, time_max.month, time_max.day, time_max.hour)
+        time_extract = Constraint(
+            time=lambda cell: time_start <= cell < time_limit)
+    return time_extract
+
+
+def extract_cube_at_time(cubes, time, time_extract):
+    """
+    Extract a single cube at a given time from a cubelist.
+
+    Args:
+        cubes (iris.cube.CubeList):
+            CubeList of a given diagnostic over several times.
+
+        time (datetime.datetime object):
+            Time at which forecast data is needed.
+
+        time_extract (iris.Constraint):
+            Iris constraint for the desired time.
+
+    Returns:
+        cube (iris.cube.Cube):
```

Comment:
At the moment, this function can return either a cube or None, so it would be useful to update this docstring to mention this.